common/lib/xmodule/xmodule/contentstore/content.py

```diff
@@ -351,27 +367,30 @@ def generate_thumbnail(self, content, tempfile_path=None, dimensions=None):
                 else:
                     im = Image.open(tempfile_path)
 
-                # I've seen some exceptions from the PIL library when trying to save palletted
-                # PNG files to JPEG. Per the google-universe, they suggest converting to RGB first.
-                im = im.convert('RGB')
+                # We use the context processor here to avoid leaking the inner file descriptor
+                # of the Image object -- this way it gets closed after we're done with using it.
+                thumbnail_file = StringIO.StringIO()
+                with im:
```

Comment:
This is unusual, but looks like a fine way to ensure the resource is closed when we have two different ways to open the resource.  Usually the with-statement opens the resource, but the important thing is that it will close the resource.  I'd prefer not reusing the "im" name like this here, since it looks like the with statement will close the newly converted image, rather than the original source image.